keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/ec2/ClusterSpec.kt

```diff
@@ -140,6 +144,13 @@ data class ClusterSpec(
   val defaults: ServerGroupSpec
     @JsonUnwrapped get() = _defaults
 
+  override val artifactReference: String?
+    @JsonIgnore get() = if (imageProvider != null && imageProvider is ReferenceArtifactImageProvider) {
```

Comment:
We can resolve the artifact reference automatically if the provider knows about references.